extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -73,30 +82,34 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory, JsonFacto
   public ExportResult<PhotosContainerResource> export(UUID jobId, TokensAndUrlAuthData authData,
       Optional<ExportInformation> exportInformation) throws IOException {
     if (!exportInformation.isPresent()) {
+      populateContainedPhotosList(jobId, authData);
       return exportAlbums(authData, Optional.empty());
+    }
+    StringPaginationToken paginationToken =
+        (StringPaginationToken) exportInformation.get().getPaginationData();
+    IdOnlyContainerResource idOnlyContainerResource =
+        (IdOnlyContainerResource) exportInformation.get().getContainerResource();
+
+    boolean containerResourcePresent = idOnlyContainerResource != null;
+    boolean paginationDataPresent = paginationToken != null;
+
+    if (!containerResourcePresent
```

Comment:
Want to add some comments about the expected flow here?

Reply:
I believe this clarifies some things.